types/src/libra_timestamp.rs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    access_path::{AccessPath, Accesses},
+    account_config,
+    language_storage::StructTag,
+};
+use move_core_types::identifier::{IdentStr, Identifier};
+use once_cell::sync::Lazy;
+use serde::{Deserialize, Serialize};
+
+static LIBRA_TIMESTAMP_MODULE_NAME: Lazy<Identifier> =
+    Lazy::new(|| Identifier::new("LibraTimestamp").unwrap());
+static LIBRA_TIMESTAMP_STRUCT_NAME: Lazy<Identifier> =
+    Lazy::new(|| Identifier::new("CurrentTimeMicroseconds").unwrap());
+
+pub fn libra_timestamp_module_name() -> &'static IdentStr {
+    &*LIBRA_TIMESTAMP_MODULE_NAME
+}
+
+pub fn libra_timestamp_struct_name() -> &'static IdentStr {
+    &*LIBRA_TIMESTAMP_STRUCT_NAME
+}
+
+pub fn libra_timestamp_tag() -> StructTag {
+    StructTag {
+        address: account_config::CORE_CODE_ADDRESS,
+        name: libra_timestamp_struct_name().to_owned(),
+        module: libra_timestamp_module_name().to_owned(),
+        type_params: vec![],
+    }
+}
+
+/// The access path where the Validator Set resource is stored.
+pub static LIBRA_TIMESTAMP_RESOURCE_PATH: Lazy<Vec<u8>> =
+    Lazy::new(|| AccessPath::resource_access_vec(&libra_timestamp_tag(), &Accesses::empty()));
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct LibraTimestampResource {
```

Comment:
the move type mapping to rust types have a lot boilerplate, can we support it by sth like #[derive(Move(metadata))]?

Reply:
I guess it's more of a question for @sblackshear / @runtian-zhou

Reply:
I think this is a must-have, and not just in Rust! I will link you to a discussion from elsewhere about where we want to go with this, but the idea is basically:

- We will provide a utility that lets you automatically extract a JSON (or similar) representation of the Move types declared in a Move bytecode module.
- Languages that are frequently used by clients (e.g., Python, JS) can autogenerate their own representations from the JSON

Since we already have a Rust representation of a Move module, we might be able to use a Rust macro that takes a CompiledModule as input and generates boilerplate like this file.

CC Rust macro guru Todd for his opinion on the feasibility/desirability of this.